### ISSUE TITLE
[VMS] Add pagination support to `getNearestEventsCardData()`

### ISSUE THREAD

**Issue by PC98** [GTBitsOfGood/gen-soln#193]
* Follow the instructions in #186, the changes for this task will be nearly identical.
* One new product requirement is that `getNearestEventsCardData()` should only return events within 20 miles of the passed in `{ lat, long }` coordinates. Check if the `near` query can be modified to take into account some sort of radius field.

**Comment by matthewbrown77** [comment]
`Near` supports a [maxDistance](https://docs.mongodb.com/manual/reference/operator/query/near/) field that we can use!

**Comment by matthewbrown77** [comment]
When you say that the changes are identical, are we still sorting and paginating by the startDate?

**Comment by PC98** [comment]
No, `getNearestEventsCardData()` won't look at the `startDate` and will continue to query based off of an Event's `address.location` and the passed in `{ lat, long }` object.

Thinking about this more, I realized there will be a minor point of difference `getNearestEventsCardData()` and `getUpcomingEventsCardData()`. You see `getNearestEventsCardData()` will need a user's location information, which we will most probably get from HTML 5's Geolocation API. This means that `getNearestEventsCardData()` will be called from the client-side every time, including the first time. `getUpcomingEventsCardData()` will be called from the server-side the first time and only subsequent calls come from the client.

However, to simplify things in this sprint, let's just implement it exactly like #186 (but again, the query will be location-based and not date based here).

**Comment by PC98** [comment]
Might also be a good idea to look at https://github.com/GTBitsOfGood/gen-soln/pull/196, the PR for #186 while doing this task

**Comment by matthewbrown77** [comment]
I’ve been using it for reference but decided wait on opening another PR until it’s merged to avoid conflicts.

**Comment by matthewbrown77** [comment]
Is there a reason we decided to use the `near` operator instead of `within`? `Near` is a restricted operator for `countDocuments()`, and I think it would be best if our location querying was consistent across the file.

**Comment by PC98** [comment]
Sure let's use within then.
